src/main/kotlin/org/partiql/scribe/sql/SqlFeatures.kt

```diff
@@ -1,136 +1,55 @@
-
-    override fun visitCatalogSymbol(node: Catalog.Symbol, ctx: ProblemCallback)  = visitChildren(node, ctx)
-
-    override fun visitCatalogSymbolRef(node: Catalog.Symbol.Ref, ctx: ProblemCallback)  = visitChildren(node, ctx)
-
-    override fun visitFn(node: Fn, ctx: ProblemCallback)  = visitChildren(node, ctx)
-
-    override fun visitIdentifier(node: Identifier, ctx: ProblemCallback)  = visitChildren(node, ctx)
-
-    override fun visitIdentifierQualified(node: Identifier.Qualified, ctx: ProblemCallback)  = visitChildren(node, ctx)
-
-    override fun visitIdentifierSymbol(node: Identifier.Symbol, ctx: ProblemCallback)  = visitChildren(node, ctx)
-
-    override fun visitPartiQLPlan(node: PartiQLPlan, ctx: ProblemCallback)  = visitChildren(node, ctx)
-
-    override fun visitRel(node: Rel, ctx: ProblemCallback)  = visitChildren(node, ctx)
-
-    override fun visitRelBinding(node: Rel.Binding, ctx: ProblemCallback)  = visitChildren(node, ctx)
-
-    override fun visitRelType(node: Rel.Type, ctx: ProblemCallback)  = visitChildren(node, ctx)
-
-    override fun visitRex(node: Rex, ctx: ProblemCallback)  = visitChildren(node, ctx)
-
-    override fun visitRexOpCallDynamicCandidate(node: Rex.Op.Call.Dynamic.Candidate, ctx: ProblemCallback) = visitChildren(node, ctx)
-
-    override fun visitRexOpCaseBranch(node: Rex.Op.Case.Branch, ctx: ProblemCallback) = visitChildren(node, ctx)
-
-    override fun visitRexOpStructField(node: Rex.Op.Struct.Field, ctx: ProblemCallback) = visitChildren(node, ctx)
-
-    override fun visitStatement(node: Statement, ctx: ProblemCallback)  = visitChildren(node, ctx)
-
-    override fun visitStatementQuery(node: Statement.Query, ctx: ProblemCallback)  = visitChildren(node, ctx)
-
-    /**
-     * This visits all of the [node]'s children without invoking the [defaultReturn] for the [node].
-     */
-    public open fun visitChildren(node: PlanNode, ctx: ProblemCallback) {
-        node.children.forEach { it.accept(this, ctx) }
+import org.partiql.plan.Action
+import org.partiql.plan.Operator
+import org.partiql.plan.OperatorVisitor
+import org.partiql.plan.Plan
+import org.partiql.scribe.ScribeContext
+import org.partiql.scribe.problems.ScribeProblem
+
+public abstract class SqlFeatures : OperatorVisitor<Unit, ScribeContext> {
+    public fun validate(
+        plan: Plan,
+        context: ScribeContext,
+    ) {
+        when (val action = plan.action) {
+            is Action.Query -> visit(action.rex, context)
+            else ->
+                context.getErrorListener().report(
+                    ScribeProblem.simpleError(
+                        ScribeProblem.UNSUPPORTED_OPERATION,
+                        "Can only transform a query statement",
+                    ),
+                )
+        }
     }
 
-    /**
-     * An [SqlFeatures] which denies all features (rel.op and rex.op) by default; thereby requiring explicit opt-in.
-     */
```

Comment:
Seems like this comment is still valid to keep, no?

Reply:
Yea, I'll add it back.